File: Lecture5/SoccerManager/Keeper.cpp
#include <stdlib.h>
#include "Keeper.h"

Keeper::Keeper():
    Player(),
    reaction(1){}
Keeper::Keeper(string n, int a, int s, int t, int m, int r):
    Player(n,a,s,t,m),
    reaction(r){}

//functions
bool Keeper::saveShot(int shot)
{
    int r=rand()%3-1;
    int ret=max(1,min(10,reaction+r));

    if(ret>=shot)
    {
        return true; //saved
    }else
    {
        return false; //SCOOOORE
    }
}

File: Lecture5/Python/SoccerManager/SoccerManager/Keeper.py
from Player import *


class Keeper(Player):
    __reaction = 1

    def __init__(self, name: str, age: int, st: int, sh: int, mo: int, re: int):
        Player.__init__(self, name, age, st, sh, mo)
        self.__reaction = re

    def save_shot(self, shot: int) -> bool:
        r = random.randint(-1, 1)
        ret = max(1, min(10, self.__reaction+r))

        if ret >= shot:
            return True
        else:
            return False
